core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,87 @@
+{% macro copy_grants() %}
+    {{ return(adapter.dispatch('copy_grants', 'dbt')()) }}
+{% endmacro %}
+
+{% macro default__copy_grants() %}
+    {{ return(True) }}
+{% endmacro %}
+
+{% macro should_revoke(existing_relation, full_refresh_mode=True) %}
+
+    {% if not existing_relation %}
+        {#-- The table doesn't already exist, so no grants to copy over --#}
+        {{ return(False) }}
+    {% elif full_refresh_mode %}
+        {#-- The object is being REPLACED -- whether grants are copied over depends on the value of user config --#}
+        {{ return(copy_grants()) }}
+    {% else %}
+        {#-- The table is being merged/upserted/inserted -- grants will be carried over --#}
+        {{ return(True) }}
+    {% endif %}
+
+{% endmacro %}
+
+{% macro get_show_grant_sql(relation) %}
+    {{ return(adapter.dispatch("get_show_grant_sql", "dbt")(relation)) }}
+{% endmacro %}
+
+{% macro default__get_show_grant_sql(relation) %}
+    show grants on {{ relation }}
+{% endmacro %}
+
+{% macro get_grant_sql(relation, grant_config) %}
+    {{ return(adapter.dispatch('get_grant_sql', 'dbt')(relation, grant_config)) }}
+{% endmacro %}
+
+{%- macro default__get_grant_sql(relation, grant_config) -%}
+    {%- for privilege in grant_config.keys() -%}
+        {%- set grantees = grant_config[privilege] -%}
+        {%- if grantees -%}
+            {%- for grantee in grantees -%}
+                grant {{ privilege }} on {{ relation }} to {{ grantee }};
+            {%- endfor -%}
```

Comment:
Caught a bug. This doesn't properly handle multiple `grantees` (whitespace). We can more efficiently provide a list of multiple grantees by joining them together as a comma-separated list: `grant ... to grantee_a, grantee_b, ...`

```suggestion
            grant {{ privilege }} on {{ relation }} to {{ grantees | join(', ' }};
```

We need testing for multiple grantees!

Reply:
made changes for both revoke and grant statements trying to test it out now! locally with what we currently have

Reply:
change made in more grant adapter tests pr